esp/esp/program/modules/base.py

```diff
@@ -548,7 +558,7 @@ def meets_deadline(extension=''):
     def meets_deadline(method):
         def _checkDeadline(moduleObj, request, tl, *args, **kwargs):
             errorpage = 'errors/program/deadline-%s.html' % tl
-            (canView, response) = _checkDeadline_helper(method, extension, moduleObj, request, tl, *args, **kwargs)
+            (canView, response) = _checkDeadline_helper(method, extension, moduleObj, request, tl, require_auth = moduleObj.require_auth(), *args, **kwargs)
```

Comment:
I'd actually even check `moduleObj.require_auth()` inside `_checkDeadline_helper`, so that you can make `meets_any_deadline` and friends work right for free and save passing it around a bit.